lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1901,19 +1897,32 @@ sub _get_uuid_string {
 
 sub get_merged_bam_to_revivify_per_lane_bam {
     my $self = shift;
-    my $merged_result = $self->get_smallest_merged_alignment_result($self->get_unarchived_merged_alignment_results);
+    my $merged_bam;
 
-    unless ($merged_result) {
-        $merged_result = $self->get_smallest_merged_alignment_result($self->get_merged_alignment_results);
-        unless ($merged_result) {
-            die $self->error_message('Failed to get archived merged result for per lane alignment '.$self->id);
+    for my $mr ($self->get_unarchived_merged_alignment_results) {
```

Comment:
This could be changed to `get_active_merged_alignment_results`.  If it filters on `$_->status eq 'active'` instead of `$_->is_archived`, it will exclude both archived and purged results.